### ISSUE TITLE
"Something went wrong" errors began surfacing recently while using the add-in - bisect does not help, we suspect an Excel for the Web issue

### ISSUE THREAD

**Issue by wh1t3cAt1k** [OfficeDev/office-js#1768]
Problem: files using our add-in, opened in Chromium-based browsers in Excel Online, started exhibiting numerous and frequent non-descriptive "something went wrong" errors during recalculation, making the add-in unusable in many cases.

The error is sporadic but its frequency seems correlated to the number of cells using our custom functions in the file.

The add-in is called [Velixo](https://appsource.microsoft.com/en-us/product/office/wa200002311?tab=overview) if it's required.

At first, I suspected a regression in our code. However, bisecting proved unfruitful and even older revisions of the add-in, which used to work just fine in production, are now SWW'ing like crazy when we check out those revisions locally.

**Sample Error Ids**:

2450e54e-d897-426e-8bfd-582588d810d3, 2021-03-31 14:51:28Z 
Happened on session id 4b5b40c3-1203-4080-b2f6-94785c7d13ef and **build** 16.0.13921.31876
Browser: **Edge** Version 89.0.774.63 (Official build) (64-bit)

An ID of an earlier error: 
5aa5b21b-ea25-488a-aa14-42a01f9072d2, 2021-03-30 12:38:13Z

(do you have telemetry to check where those errors originated?)

Setup:

- Excel Online
- Chrome, Edge, or Firefox - **never** observed on Safari or Excel for Mac.
- A file containing >10 000 custom functions. Unfortunately cannot share the report because it is a P&L statement from a customer.

We don't really understand what is going on, however we have some anecdotal evidence that the errors began surfacing after this feature had been announced: https://insider.office.com/en-us/blog/collaborate-with-confidence-with-show-changes-in-excel-web?ocid=usoc_TWITTER_M365_spl100001892525966 . Does that build 16.0.13921.31876 even have this feature?

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: **Office on the Web**
* Host [Excel, Word, PowerPoint, etc.]: **Excel**
* Office version number: 16.0.13921.31876
* Operating System: Mac OS X / Windows 10 (regardless)
* Browser (if using Office on the web): **Edge** Version 89.0.774.63

**Comment by wh1t3cAt1k** [comment]
Another error instance occurred just now on the same set-up (Edge Chromium).

This time, I replaced all my custom function definitions with 

```typescript
await Promise.resolve();
return 42;
```

Error Id: 0a2fbefb-becc-4be9-b426-60d75694d48c, 2021-03-31 18:51:54Z

**Comment by wh1t3cAt1k** [comment]
More experiments: this time, I threw off the Redux store, sagas, epics, the UI – everything. 

I left just an `h1` with an `onClick` handler that awaits `Office.onReady()` and then invokes `Excel.run()` to do just one operation: `context.workbook.application.calculate('Full')` .

That's it.

Enough clicks at this h1 (sometimes up to twenty, diligently waiting for the previous calculation to end) and here goes "something went wrong".

Error Id: 595d7711-d064-42b3-805d-73382a73ee9a, 2021-03-31 19:01:22Z

TLDR:

- A file with multiple cells containing custom functions (20 000 cells and some diagrams based on those)
- All custom functions stubbed out to asynchronously `return 42` after yielding.
- The application is stripped of everything, it's just a single header with an `onClick` handler that invokes full calculation in the workbook.

Eventually we hit this error for no apparent reason - the debugger does not stop on, or log, any promise rejections.

**Comment by Rick-Kirkham** [comment]
@lumine2008  and @grangeryy can you please take a look at this?

**Comment by wh1t3cAt1k** [comment]
Thanks for the prompt response @Rick-Kirkham , I appreciate your speed guys :)

On my end, I seem to have a pretty minimal set-up of an add-in that reproduces the issue.

My index file:

```typescript
require('../functions/functions');

import '@fluentui/react/dist/css/fabric.min.css';
import React from 'react';
import ReactDOM from 'react-dom';
import { App } from './components/app/app';

ReactDOM.render(<App />, document.getElementById('container'));
```

App component:

```typescript
import { ExcelRecalculationService } from 'packages/addin/src/services/excel-interop/excel-recalculation-service';
import React from 'react';

const App: React.FunctionComponent = () => (
    <div>
        <h1
            onClick={async (): Promise<void> => {
                await ExcelRecalculationService.recalculateFull();
            }}
        >
            CLICK ME TO GET FRUSTRATED.
        </h1>
    </div>
);

export { App };
```

Excel recalculation service:

```typescript
export class ExcelRecalculationService {
    public static readonly recalculateFull = async (): Promise<void> => {
        await Office.onReady();

        return Excel.run(
            async (context): Promise<void> => {
                context.workbook.application.calculate('Full');
                await context.sync();
            }
        );
    };
}
```

and, finally, my stubbed out custom function definitions (functions.ts imported from index - we are using shared runtime):

```typescript
import { CustomFunctionIdentifier } from './custom-function-identifier';

const queueFunctionCall = async <TResult>(): Promise<TResult> => {
    await Promise.resolve();
    return 42 as never;
};

Object.keys(CustomFunctionIdentifier).forEach(name => {
    CustomFunctions.associate(name, async () => queueFunctionCall());
});
```

The key seems to be having maaaaany custom function cells in the file, ideally with some dependent diagrams and such.

**Comment by lumine2008** [comment]
This issue looks like related to webview,  It doesn't seem related to Excel JS API @Rick-Kirkham, would you please also loop contact from web view?

**Comment by wh1t3cAt1k** [comment]
Thanks @lumine2008 @Rick-Kirkham , please keep us posted. The issue does not seem transient, I encountered it again many times last week.

**Comment by Rick-Kirkham** [comment]
@smaremanda @jeremy-msft Can you find someone who can help with this?

**Comment by grangeryy** [comment]
Bug 5026687 is created to track this issue and we'll investigate soon.

**Comment by xiaochunms** [comment]
@wh1t3cAt1k Not sure if you are still hitting the error as such. If so, may you please share the latest Error Id? The previous Error Id was out of date and a latest one will be much helpful for the investigation. Sorry about that and thanks a lot for your understanding.

**Comment by wh1t3cAt1k** [comment]
Hi guys,

We haven't observed this error recently. Not sure what it was. Thanks!

**Comment by grangeryy** [comment]
@wh1t3cAt1k thanks for the info! Will close the issue for temporary and feel free to reopen it once the error shows up again.
